src/library_pthread.js

```diff
@@ -321,6 +321,10 @@ var LibraryPThread = {
             url: currentScriptUrl,
             buffer: HEAPU8.buffer,
             tempDoublePtr: tempDoublePtr,
+            TOTAL_MEMORY: TOTAL_MEMORY,
+            STATICTOP: STATICTOP,
+            DYNAMIC_BASE: DYNAMIC_BASE,
+            DYNAMICTOP_PTR: DYNAMICTOP_PTR,
```

Comment:
Here I did a slightly unrelated change. Before, the runtime in pthreads would compute these variables itself, and since it's executing the same .js files and these are static, pthreads would come to the same "conclusion" for these values even when they recompute these four fields. It is conceptually nicer to have pthreads never compute any of these even though they are static, but always have the main runtime pass the values over when constructing the threads workers. so that they will explicitly stay in sync (if something might change in the future).